src/components/Dropdown/Dropdown.tsx

```diff
@@ -122,10 +119,10 @@ export const Dropdown = ({
 
     useEffect(() => {
         setItems(list);
-        if (shouldClearValue && !list.some(item => item.text === getInputProps().value)) {
-            const result = value ? list.find(item => item.value && item.value === value)?.text : '';
-            setInputValue(reactNodeToString(result) ?? '');
-        }
+
+        const result = value ? list.find(item => item.value && item.value === value)?.text : '';
+        setInputValue(reactNodeToString(result) ?? '');
+
```

Comment:
won't these affect other pages that are using this component?

Reply:
i have checked where else we used it and its only in the BusinessHoursModal TimeDropdown component. I tested it and it seemed like it was working

Reply:
ok after checking, i need this back, its breaking the modal, ill find another way to fix it